constraint/pkg/client/client_test.go

```diff
@@ -606,3 +606,78 @@ func TestRemoveConstraint(t *testing.T) {
 		})
 	}
 }
+
+func TestAllowedDataFields(t *testing.T) {
+	inventoryTempl := createTemplate(name("fake"), crdNames("Fake"), targets("h1"))
+	inventoryTempl.Spec.Targets[0].Rego = `
+package something
+
+violation[{"msg": "msg"}] {
+	data.inventory = "something_else"
+}
+`
+
+	tc := []struct {
+		Name          string
+		Handler       TargetHandler
+		Template      *templates.ConstraintTemplate
+		ErrorExpected bool
+		InvAllowed    bool
+	}{
+		{
+			Name:          "Inventory Not Used",
+			Handler:       &badHandler{Name: "h1", HasLib: true},
+			Template:      createTemplate(name("fakes"), crdNames("Fakes"), targets("h1")),
+			ErrorExpected: false,
+		},
+		{
+			Name:          "Inventory Used",
+			Handler:       &badHandler{Name: "h1", HasLib: true},
+			Template:      inventoryTempl,
+			ErrorExpected: true,
+		},
+		{
+			Name:          "Inventory Used But Allowed",
+			Handler:       &badHandler{Name: "h1", HasLib: true},
+			Template:      inventoryTempl,
+			ErrorExpected: false,
+			InvAllowed:    true,
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.Name, func(t *testing.T) {
+			d := local.New()
+			b, err := NewBackend(Driver(d))
+			if err != nil {
+				t.Fatalf("Could not create backend: %s", err)
+			}
+			f := AllowedDataFields()
+			if tt.InvAllowed {
+				f = AllowedDataFields("inventory")
```

Comment:
can we add another test for if `f = AllowedDataFields("notallowed")`? and make sure NewClient returns an empty c.allowedDataFields

Reply:
done